### PR TITLE
SDIT-2319 Add params to findPrisonersByName and ui improvements

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: integration_tests/e2e/search.cy.ts [hunk 1]
```diff
@@ -17,11 +17,32 @@ context('Sign In', () => {
     prisonerSearchPage.searchSelectRadioButton('Name/age').should('be.checked')
   })
 
+  it('Will display search data entered when the page is reloaded', () => {
```
Comment (karenmillermoj) [inline integration_tests/e2e/search.cy.ts:20]: next step add more tests

File: server/routes/search/searchController.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  async postSearch(req: Request, res: Response): Promise<void> {
+    logger.debug('post search /')
+    // TODO Validate form
+
+    const prisonerSearchForm = req.body
+    req.session.prisonerSearchForm = { ...trimForm(req.body) }
+    logger.debug('searchform is', req.session.prisonerSearchForm)
+
+    if (prisonerSearchForm.searchType === 'name') {
```
Comment (karenmillermoj) [inline server/routes/search/searchController.ts:30]: This looks a bit naff - can change to switch statement - not sure of a neater solution

File: server/routes/search/searchController.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      const { content } = await this.historicalPrisonerService.findPrisonersByAddressTerms(
+        req.user.token,
+        prisonerSearchForm,
+      )
+      res.render('pages/search', { searchResults: content, form: req.session.prisonerSearchForm })
+    }
+  }
+
+  private static toPrisonersByName(form: PrisonerSearchForm): FindPrisonersByName {
+    // TODO Format all fields correctly
```
Comment (karenmillermoj) [inline server/routes/search/searchController.ts:50]: wip

File: server/routes/search/searchRouter.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import type { RequestHandler, Router } from 'express'
```
Comment (karenmillermoj) [inline server/routes/search/searchRouter.ts:1]: pulled into separate file

File: server/services/historicalPrisonerService.ts [hunk 5]
```diff
@@ -1,33 +1,42 @@
...
-    nameSearchForm: {
-      forename?: string
-      surname?: string
-      dateOfBirth?: string
-      page?: number
-    },
-  ): Promise<PagedModelPrisoner> {
-    return this.historicalPrisonerApiClient.findPrisonersByNameOrAge(token, nameSearchForm)
+    prisonersByNameForm: FindPrisonersByName,
+  ): Promise<PagedModelPrisonerSearchDto> {
```
Comment (karenmillermoj) [inline server/services/historicalPrisonerService.ts:20]: Have removed the apiclient - not sure if we need both service and client calls as a lot of the work is actually done in the controller.  Service was just acting like the apiclient

File: server/utils/trim.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
+export default function trimForm<T extends Record<string, unknown>>(form: T): T {
+  return Object.keys(form).reduce((acc, curr) => {
```
Comment (karenmillermoj) [inline server/utils/trim.ts:2]: will see if there is anything internal that I can use - saw other services using this

File: server/views/pages/search.njk [hunk 7]
```diff
@@ -1,208 +1,90 @@
...
+{% from "govuk/components/tag/macro.njk" import govukTag %}
+{% from "moj/components/filter/macro.njk" import mojFilter %}
 
 {% extends "../partials/layout.njk" %}
 
 {% set pageTitle = "Usage" %}
 {% set mainClasses = "app-container govuk-body" %}
 
 {% block content %}
-
```
Comment (karenmillermoj) [inline server/views/pages/search.njk:17]: separated from into its own file

File: server/views/pages/search.njk [hunk 8]
```diff
@@ -1,208 +1,90 @@
...
-                  <div class="govuk-grid-column-three-quarters">
-                    <div class="govuk-body-lead">
-                      {{ prisoner.primaryForename1 }} {{ prisoner.prismarySurname }} &middot; {{ prisoner.primaryBirthDate | formatDate }}
-                    </div>
-                    <div class="govuk-body-m">
-                      Prison no. {{ prisoner.prisonNumber }} &middot; Reception {{ prisoner.receptionDate | formatDate }}
-                    </div>
-                    <div class="govuk-body-s">
-                      Add to shortlist
+                {{ mojPagination(pagination) }}
```
Comment (karenmillermoj) [inline server/views/pages/search.njk:45]: wip - paging not yet set up and still needs tidy up

File: server/routes/search/searchController.ts [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+export default class SearchController {
+  constructor(
+    private readonly historicalPrisonerService: HistoricalPrisonerService,
+    private readonly auditService: AuditService,
+  ) {}
+
+  async getSearch(req: Request, res: Response): Promise<void> {
+    logger.debug('search /')
+    await this.auditService.logPageView(Page.SEARCH, { who: res.locals.user.username, correlationId: req.id })
```
Comment (petergphillips) [inline server/routes/search/searchController.ts:17]: not sure it is worth logging that they are on the search page

File: server/routes/search/searchController.ts (hunk 2 above)
Comment (petergphillips) [reply server/routes/search/searchController.ts:30]: I guess you could move the logic down to the service?

File: server/services/healthCheck.ts [hunk 10]
```diff
@@ -44,6 +44,7 @@ function gatherCheckInfo(aggregateStatus: Record<string, unknown>, currentStatus
 
 const apiChecks = [
   service('hmppsAuth', `${config.apis.hmppsAuth.url}/health/ping`, config.apis.hmppsAuth.agent),
+  service('historicalPrisonerApi', `${config.apis.historicalPrisonerApi.url}/health/ping`, config.apis.hmppsAuth.agent),
```
Comment (petergphillips) [inline server/services/healthCheck.ts:47]: weird sharing the auth agent?

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**